### PR TITLE
Sebastian

### PR DESCRIPTION
**Pull request by sebmader**
We updated the tests for #270.
They overlap quite a bit with the ones of #367 that we discussed today but I feel, it makes sense to stick with the original plan of implementing backwards acceleration the same way as forward acceleration is implement right now, and change it together with the split of acceleration and move.

### CODE REVIEW

**Comment by sebmader** [comment]
I don't know where all these changes are coming from....?!
Will look at it next time..

**Comment by richelbilderbeek** [comment]
@sebmader I guess some changes come from the merge of develop to the sebastian branch. I guess the review only needs to focus on the acceleration changes, on not the other changes that have reached develop :-)

**Comment by TheoPannetier** [comment]
Nope, let's not merge from `sebastian` to `master` %)

File: player.cpp [hunk 1]
```diff
@@ -584,26 +590,45 @@ void test_player() //!OCLINT tests may be long
     // A players goes ?right/?up upon acceleraton
     {
         player p_forward;
+        coordinate c_before = get_coordinate(p_forward);
         p_forward.accelerate();
```
Comment (TheoPannetier) [inline player.cpp:594]: A bit confued here: haven't we agreed that `accelerate()` should *not* move the player?

File: player.cpp [hunk 2]
```diff
@@ -584,26 +590,45 @@ void test_player() //!OCLINT tests may be long
...
         player p;
+        coordinate c_before = get_coordinate(p);
         p.accelerate();
-        assert(get_dx(p) > 0.0); // Get the delta x, maybe needs to be added
-        assert(get_dy(p) > 0.0); // Get the delta x, maybe needs to be added
+        coordinate c_inbetween = get_coordinate(p);
+        double dx_a = get_x(c_inbetween) - get_x(c_before);
+        double dy_a = get_y(c_inbetween) - get_y(c_before);
+        assert(dx_a > 0.0); // Get the delta x
+        assert(dy_a > -0.00000000001 && dy_a < 0.00000000001);
```
Comment (TheoPannetier) [inline player.cpp:622]: That bit is redundant with the block just above isn't it?

File: player.cpp [hunk 3]
```diff
@@ -584,26 +590,45 @@ void test_player() //!OCLINT tests may be long
...
         p.acc_backward();
         p.acc_backward();
-        assert(get_dx(p) < 0.0); // Signs should flip
-        assert(get_dy(p) < 0.0); // Signs should flip
+        coordinate c_after = get_coordinate(p);
+        double dx_b = get_x(c_after) - get_x(c_inbetween);
+        double dy_b = get_y(c_after) - get_y(c_inbetween);
+        assert(dx_b < 0.0); // Signs should flip
+        // it should not change in the y direction if the assumption about initial direction is correct
+        assert(dy_b > -0.00000000001 && dy_b < 0.00000000001);
```
Comment (TheoPannetier) [inline player.cpp:631]: Oh that's brilliant, I like this being checked for

File: player.cpp (hunk 2 above)
Comment (sebmader) [reply player.cpp:622]: What do you mean? Pls be more specific :)

**Pull request closed without merging**